addons/auth_signup/controllers/main.py

```diff
@@ -50,11 +50,15 @@ def web_auth_signup(self, *args, **kw):
             raise werkzeug.exceptions.NotFound()
 
         if 'error' not in qcontext and request.httprequest.method == 'POST':
-            try:
-                self.do_signup(qcontext)
-                return super(AuthSignupHome, self).web_login(*args, **kw)
-            except (SignupError, AssertionError), e:
-                qcontext['error'] = _(e.message)
+            if request.env["res.users"].sudo().search(
```

Comment:
Probably move the check inside the `except` block. Try registering and if it fails, try to understand why.